presto-main/src/main/java/io/prestosql/operator/HashSemiJoinOperator.java

```diff
@@ -131,6 +132,7 @@ public HashSemiJoinOperator(
             Optional<Integer> probeHashChannel,
             MemoryTrackingContext memoryTrackingContext)
     {
+        this.channelSetFuture = requireNonNull(channelSetFuture, "channelSetFuture is null").getChannelSet();
```

Comment:
you also need to pass `channelSetFuture` to `PagesBuffer` constructor, see: `io.prestosql.operator.LookupJoinOperator#LookupJoinOperator`